scripts/publish.js

```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+const fs = require('fs');
+const { execSync } = require('child_process');
+const { getPackageJSON } = require('./helpers');
+const COMPONENTS_PATH = path.join(__dirname, '../components');
+
+function getPublicVersion(package) {
+    return execSync(`npm view ${package} version`, {encoding: 'utf8'}).replace('\n', '');
+}
+
+function hasAnUpdate(component) {
+    const packageJSON = getPackageJSON(component);
+    if (!packageJSON) return false;
+
+    const name = packageJSON.name;
+
+    // if a component doesn't exist in the registry then it must be published
+    if (!execSync(`npm search ${name} --json`).length) return true;
+
+    const localVersion = packageJSON.version;
+    const publicVersion = getPublicVersion(name);
+
+    if(localVersion !== publicVersion) return true;
+    return false;
+}
+
+function main() {
+    const components = fs.readdirSync(COMPONENTS_PATH);
+    for (let component of components) {
+        if(hasAnUpdate(component)) {
+            console.log(`Component ${component} has new version - ${localVersion}, current is ${publicVersion}.`);

```

Comment:
There is no `localVersion` and `publicVersion` in this scope.

Reply:
Fixed. Moved to shouldUpdate.
I added the log for info purposes, do you think we should keep it?
Also should we add a message after successful publish? - I think we do - added, let me know if you disagree.